internal/metrics/submitter/main.go

```diff
@@ -0,0 +1,103 @@
+package submitter
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"os"
+	"time"
+
+	"github.com/stackrox/image-prefetcher/internal/metrics/gen"
+
+	"github.com/cenkalti/backoff/v4"
+)
+
+type Submitter struct {
+	c      chan *gen.Result
+	done   chan struct{}
+	client gen.MetricsClient
+	logger *slog.Logger
+}
+
+func NewSubmitter(logger *slog.Logger, client gen.MetricsClient) *Submitter {
+	return &Submitter{
+		c:      make(chan *gen.Result, 1),
+		done:   make(chan struct{}),
+		client: client,
+		logger: logger,
+	}
+}
+
+func (s *Submitter) Run(ctx context.Context) {
```

Comment:
How about returning an error?

Reply:
It wouldn't be terribly useful since Run() is called in a goroutine which drops the error on the floor anyway. And Run already does retries and logging.